### ISSUE TITLE
Unable to create BGP Session object using pynetbox, when peer_group is supplied in the args

### ISSUE THREAD

**Issue by pspdbx** [k01ek/netbox-bgp#46]
**NetBox version**
2.11.2

**Describe the bug**
Unable to create BGP Session object using pynetbox, when peer_group is supplied in the args.

**To Reproduce**
nb_client in the output below is my instance of pynetbox.api

`In [943]: object_dict
Out[943]:
{'import_policies': [51, 66, 18],
 'export_policies': [60, 18],
 'description': 'LEVEL3',
 'peer_group': 8,
 'local_address': {'address': '4.71.253.6/30'},
 'remote_address': {'address': '4.71.253.5/30'},
 'local_as': {'number': 12345},
 'remote_as': {'number': 3356},
 'name': '3356_TRANSIT_LEVEL3_mydevice',
 'device': {'name': 'mydevice'}}

In [944]: nb_staging.plugins.bgp.session.create(object_dict)
---------------------------------------------------------------------------
RequestError                              Traceback (most recent call last)
<ipython-input-944-5e86e2d8ef9b> in <module>()
----> 1 nb_staging.plugins.bgp.session.create(object_dict)

/netbox_dev/nsot-ipython.runfiles/__main__/pip/pynetbox/pynetbox-cpython-38/lib/pynetbox/core/endpoint.py in create(self, *args, **kwargs)
    302         """
    303
--> 304         req = Request(
    305             base=self.url,
    306             token=self.token,

/netbox_dev/nsot-ipython.runfiles/__main__/pip/pynetbox/pynetbox-cpython-38/lib/pynetbox/core/query.py in post(self, data)
    385         :Returns: Dict containing the response from NetBox's API.
    386         """
--> 387         return self._make_call(verb="post", data=data)
    388
    389     def delete(self):

/netbox_dev/nsot-ipython.runfiles/__main__/pip/pynetbox/pynetbox-cpython-38/lib/pynetbox/core/query.py in _make_call(self, verb, url_override, add_par
ams, data)
    280                 raise ContentError(req)
    281         else:
--> 282             raise RequestError(req)
    283
    284     def concurrent_get(self, ret, page_size, page_offsets):

RequestError: The request failed with code 500 Internal Server Error: {'error': "argument of type 'BGPPeerGroup' is not iterable", 'exception': 'TypeError', 'netbox_
version': '2.11.2', 'python_version': '3.8.8+'}`

**Expected behavior**
The BGP Session object gets created

**Screenshots**
n/a - the traceback above should be sufficient. 

**Additional context**
I have also tried changing 
`object_dict['peer_group'] = {'name': 'TRANSIT'}` 
but I still get the same error. Neither referencing related object by a dictionary of attributes nor by its ID seem to work. 

If I remove the 'peer_group' key from object_dict altogether, the BGP session object gets created successfully. However, thats not what I would like. I'd like to relate the session to a peer group.

**Comment by pspdbx** [comment]
I would add that the problem is not just during creation. If I create the BGP Session object without relating it any Peer Group, and later try to update the Session object, I get the same error.

**Comment by pspdbx** [comment]
Both creating and updating a BGP Session with Peer Group, works fine over the GUI. The problem only seems to be when pynetbox is used to either create/update BGP session with a related peer_group. Also, removing a related peer_group from a BGP session works fine over the pynetbox client/API.

**Comment by k01ek** [comment]
I have dug into this bug and found the reason. There is some problem in rest_framework with nested serializer and unique together validation. I found the workaround to fix the problem: remove validation for the nested serializer, but I afraid it can have some side effects. You can use this fix https://github.com/k01ek/netbox-bgp/tree/quick-fix-46 
I will look for a more elegant solution.

**Comment by pspdbx** [comment]
Thank you @k01ek

**Comment by pspdbx** [comment]
What is the impact of rolling this fix into the master branch @k01ek  ? After all, the change is restricted only to BGP Peer Group Serializer

**Comment by petercrocker** [comment]
Just ran in to this as well. I've just avoided peer groups for the short term until this can be further tested and rolled in to master.

**Comment by pspdbx** [comment]
yeah, I'm doing the same for the time being. Can't wait for it to get rolled into the master branch

**Comment by k01ek** [comment]
Sorry guys, I was on vacation. Will update it soon.
